src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +56,11 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.range(0, numbers.size())
+                .map(index -> index % 2 == 1 ? numbers.get(index) - 1 : numbers.get(index))
+                .filter(element -> element % 2 == 1)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
NoSuchElementException is thrown by default. You do not need to write it in orElseThrow parameter